httomo/wrappers_class.py

```diff
@@ -0,0 +1,170 @@
+
+        Returns:
+            xp.ndarray: a numpy or cupy array of the reconstructed data
+        """
+        # for 360 degrees data the angular dimension will be truncated while angles are not.
+        # Truncating angles if the angular dimension has got a different size
+        angular_dim_size = np.size(data, 0)
+        if np.size(data, 0) != len(angles_radians):
+            angles_radians = angles_radians[0:angular_dim_size]
+                
+        data = getattr(self.module, method_name)(data, angles_radians, **params)
+        return data   
+    
+class tomopy_wrapper(wrappers):
+    """A class that wraps TomoPy functions for httomo
+    """
+    def __init__(self,
+                 module_name: str = None,
+                 function_name: str = None,
+                 method_name: str = None):      
+        super().__init__(module_name, function_name, method_name)
+        
+        self.wrapper_method = super(tomopy_wrapper, self)._execute_generic
+        self.function_name = function_name
+        self.method_name = method_name
+        if module_name == "misc":
+            from tomopy import misc
+            self.module = getattr(misc, function_name)
+        if module_name == "prep":
+            from tomopy import prep
+            self.module = getattr(prep, function_name)
+            if function_name == "normalize":
+                func = getattr(self.module, method_name)
+                sig_params = signature(func).parameters
+                if 'dark' in sig_params and 'flat' in sig_params:
+                    self.wrapper_method = super(tomopy_wrapper, self)._execute_normalize
+        if module_name == "recon":
+            from tomopy import recon
+            # a workaround to get the correct module, TomoPy incosistency?
+            from importlib import import_module
+            recon = import_module('tomopy.recon')
+            self.module = getattr(recon, function_name)
+            if function_name == "algorithm":
+                self.wrapper_method = super(tomopy_wrapper, self)._execute_reconstruction
+
+class httomolib_wrapper:
+    """A class that wraps httomolib functions for httomo
+    """
+    def __init__(self,
+                 module_name: str,
+                 function_name: str,
+                 comm: Comm):
+        # set the GPU id
+        self.comm = comm
+        if gpu_enabled:
+            self.num_GPUs = xp.cuda.runtime.getDeviceCount()
+            self.gpu_id = int(comm.rank / comm.size * self.num_GPUs)
+        if module_name == "misc":
+            from httomolib import misc
+            self.module = getattr(misc, function_name) # imports the module
+        elif module_name == "prep":
+            from httomolib import prep
+            self.module = getattr(prep, function_name) # imports the module
+        elif module_name == "recon":
+            from httomolib import recon
+            self.module = getattr(recon, function_name) # imports the module
```

Comment:
I'd handle imports at the top of the file or use `import_module`.
```suggestion
        if module_name in ["misc", "prep", "recon"]:
            from importlib import import_module
            self.module = getattr(import_module("httomolib." + module_name), function_name)
```